### ISSUE TITLE
Add a new `size` field of the `tar_file` class and the metadata

### ISSUE THREAD

**Issue by wlandau** [wlandau/targets#180]
## Prework

* [x] Read and agree to the [code of conduct](https://github.com/wlandau/targets/blob/master/CODE_OF_CONDUCT.md) and [contributing guidelines](https://github.com/wlandau/targets/blob/master/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/wlandau/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.

## Description

Metdata like `bytes` gets corrupted when we store it because `_targets/meta/meta` is a text file. We need a new `size` field to help keep track of when files change is such a way that avoids unnecessary re-hashing on large files.

*WARNING*: this change will probably invalidate workflows completed with earlier versions of `targets`.

Despite the interruption, I think we need this because it will improve the accuracy and efficiency with which `targets` tracks files. In addition, this package has no CRAN release yet, so this is still a permissible time for breaking changes.

**Comment by wlandau** [comment]
Finished implementation, PR forthcoming. This puts us on much more solid footing (comparing hashes instead of doing messy floating point comparisons with ad hoc tolerance levels). And the back compatibility error message is very informative. So glad I caught this before rOpenSci review and CRAN.

**Comment by wlandau** [comment]
Fixed in #181.

**Comment by mattwarkentin** [comment]
FWIW, I ran into this error. I think the error message could use a minor update:
```
Error: callr subprocess failed: invalid header in _targets/meta/meta
  found: name|type|data|command|depend|seed|path|bytes|time|format|iteration|parent|children|seconds|warnings|error
  expected: name|type|data|command|depend|seed|path|time|size|bytes|format|iteration|parent|children|seconds|warnings|error
Probably because of a breaking change in the targets package.
Either the data store with tar_destroy() and rerun tar_make() or downgrade the targets package to an earlier version.
```

My suggestion would be to add some whitespace after `found: ` to pad it to the same length as `expected: ` so that the first character of the bar-separated lists align. Makes it easier to see whats different between the two lists.

Also: Either ___destroy___ the data store with tar_destroy() and rerun tar_make() or downgrade the targets package to an earlier version.

**Comment by wlandau** [comment]
Good ideas, I agree. Implemented in 70a1856796d25b2563a4fb44f75d689f81543fd3.
